src/app/public/modules/modal/modal-instance.ts

```diff
@@ -11,48 +11,106 @@ import {
   SkyModalBeforeCloseHandler
 } from './modal-before-close-handler';
 
+// TODO: this class won't show in the generated docs until this work is done:
+// https://github.com/blackbaud/skyux-docs-tools/issues/30
+
+/**
+ * Allows you to close the modal and return data from the launched modal.
+ */
 export class SkyModalInstance {
-  public componentInstance: any;
 
+  /**
+   * An event that the modal instance emits when it is about to close.
+   * It emits a `SkyModalBeforeCloseHandler` object with a `closeModal` method
+   * that closes the modal. If a subscription exists for this event,
+   * the modal does not close until the subscriber calls the `closeModal` method.
+   */
   public get beforeClose(): Observable<SkyModalBeforeCloseHandler> {
     return this._beforeClose;
   }
 
+  /**
+   * An event that the modal instance emits when it closes.
+   * It emits a `SkyCloseModalArgs` object with a `data` property that includes
+   * data passed from users on close or save and a `reason` property that indicates
+   * whether the modal was saved or closed without saving.
+   * The `reason` property accepts any string value.
+   * Common examples include `cancel`, `close`, and `save`.
+   */
   public get closed(): Observable<SkyModalCloseArgs> {
     return this._closed;
   }
 
+  /**
+   * An event that the modal instance emits when users click
+   * the <i class="fa fa-question-circle" aria-hidden="true"></i> button.
+   * If a `helpKey` parameter was specified, the `helpOpened` event broadcasts the `helpKey`.
+   */
   public get helpOpened(): Observable<string> {
     return this._helpOpened;
   }
 
+  /**
+   * A direct reference to the provided component's class.
+   */
+  public componentInstance: any;
+
   private _beforeClose = new Subject<SkyModalBeforeCloseHandler>();
 
   private _closed = new Subject<SkyModalCloseArgs>();
 
   private _helpOpened = new Subject<string>();
 
-  public close(result?: any, reason?: string, ignoreBeforeClose?: boolean) {
+  /**
+   * Closes the modal instance.
+   * @param result Specifies an object that will be emitted to
+   * subscribers of the `closed` event of the modal instance.
+   * The `SkyModalInstance` provider can be injected into a component's constructor
+   * so that this `close` function can be called from a button in the `sky-modal-footer`.
+   * @param reason Specifies the reason for the modal closing, with the default reason of `close`.
+   * @param ignoreBeforeClose Ignores the modal instance's `beforeClose` event.
```

Comment:
Since this is a boolean value, I think `Ignores` should be `Indicates whether to ignore` for consistency with how we document boolean values.